File: externals/Async-CTP/samples/CS/Windows/Phone/Concert/Finder/Concert/Finder/Model/Artist.cs
using System;
using System.ComponentModel;
using System.Net;
using System.Xml.Serialization;

namespace ConcertFinder.Model
{
    /// <summary>
    /// Represents an artist.
    /// </summary>
    public class Artist : ISearchable, IEquatable<Artist>
    {
        /// <summary>
        /// The ID of the artist.
        /// </summary>
        public string Id { get; set; }

        /// <summary>
        /// The name of the artist.
        /// </summary>
        public string Name
        {
            get { return _Name; }
            set
            {
                _Name = HttpUtility.HtmlDecode(value);
            }
        }
        private string _Name;

        /// <summary>
        /// The image for the artist.
        /// </summary>
        [XmlIgnore]
        public Uri Image
        {
            get { return !String.IsNullOrEmpty(_Image) ? new Uri(_Image) : null; }
            set { _Image = value != null ? value.OriginalString : null; }
        }
        [EditorBrowsable(EditorBrowsableState.Never)]
        public string _Image { get; set; }

        /// <summary>
        /// The Uri of the artist.
        /// </summary>
        [XmlIgnore]
        public Uri Uri
        {
            get { return !String.IsNullOrEmpty(_Uri) ? new Uri(_Uri) : null; }
            set { _Uri = value != null ? value.OriginalString : null; }
        }
        [EditorBrowsable(EditorBrowsableState.Never)]
        public string _Uri { get; set; }

        #region IEquatable

        /// <summary>
        /// Override for Equals.
        /// </summary>
        public bool Equals(Artist Artist)
        {
            var other = Artist as Artist;

            return other != null && other.Id.Equals(this.Id);
        }

        #endregion
    }
}

File: externals/Async-CTP/samples/VB/Windows/Phone/Concert/Finder/Concert/Finder/Model/Artist.vb
Imports System
Imports System.ComponentModel
Imports System.Net
Imports System.Xml.Serialization

Namespace ConcertFinder.Model

    Public Class Artist
        Implements ISearchable

        ''' <summary>
        ''' The ID of the artist.
        ''' </summary>
        Public Property Id As String Implements ISearchable.Id

        ''' <summary>
        ''' The name of the artist.
        ''' </summary>
        Public Property Name As String Implements ISearchable.Name
            Get
                Return _Name
            End Get
            Set(value As String)
                _Name = HttpUtility.HtmlDecode(value)
            End Set
        End Property
        Private _Name As String

        ''' <summary>
        ''' The image for the artist.
        ''' </summary>
        <XmlIgnore()>
        Public Property Image As System.Uri Implements ISearchable.Image
            Get
                Return If(Not [String].IsNullOrEmpty(_Image), New Uri(_Image), Nothing)
            End Get
            Set(value As System.Uri)
                _Image = If(value IsNot Nothing, value.OriginalString, Nothing)
            End Set
        End Property
        <EditorBrowsable(EditorBrowsableState.Never)>
        Public Property _Image As String Implements ISearchable._Image

        ''' <summary>
        ''' The Uri of the artist.
        ''' </summary>
        <XmlIgnore()>
        Public Property Uri As Uri Implements ISearchable.Uri
            Get
                Return If(Not [String].IsNullOrEmpty(_Uri), New Uri(_Uri), Nothing)
            End Get
            Set(value As Uri)
                _Uri = If(value IsNot Nothing, value.OriginalString, Nothing)
            End Set
        End Property
        <EditorBrowsable(EditorBrowsableState.Never)>
        Public Property _Uri As String Implements ISearchable._Uri

        ''' <summary>
        ''' Override for Equals.
        ''' </summary>
        Public Overrides Function Equals(obj As Object) As Boolean
            Dim other = TryCast(obj, Artist)
            Return other IsNot Nothing AndAlso other.Id.[Equals](Me.Id)
        End Function

    End Class

End Namespace